File: 14_Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int strNumbers = strs.size();
        int j = 0;
        if (strNumbers==0) {
            return "";
        } else if (strNumbers==1) {
            return strs[0];
        } else {
            int firstWordLen = strs[0].length();
            int i = 1;
            while (strs[0][j]==strs[i][j] && strs[0][j]!='\0') {
                i = ( i + 1) % strNumbers;
                if (i==0) {
                    j++;
                    i++;
                }
            }
        }
        return strs[0].substr(0, j);
    }
};

File: 14_Longest_Common_Prefix.py
#Horizontal scanning approach, TC: O(S(sum of length of all the wors)), SC: O(1)
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not len(strs):
            return ""
        prefix = strs[0]
        for i in range(1, len(strs)):
            while not strs[i].startswith(prefix):
                prefix = prefix[:-1]
            if len(prefix) == 0:
                return ""
        return prefix
        